app/src/routes/adminRoute.jsx

```diff
@@ -12,13 +16,28 @@ export const adminRoute = (Component) =>
       authorized: isAuthorizedSelector(state),
       accountRole: userAccountRoleSelector(state),
       projectId: activeProjectSelector(state),
+      isAdminAccess: isAdminAccessSelector(state),
     }),
     (dispatch) => ({
       redirect: (projectId) =>
         dispatch(rfrRedirect({ type: PROJECT_DASHBOARD_PAGE, payload: { projectId } })),
+      setAdminAccess: () => dispatch(setAdminAccessAction()),
     }),
-  )(({ authorized, accountRole, projectId, redirect, ...otherProps }) => {
-    if (authorized && accountRole === ADMINISTRATOR) return <Component {...otherProps} />;
-    redirect(projectId);
-    return null;
-  });
+  )(
+    ({
+      authorized,
+      accountRole,
+      projectId,
+      isAdminAccess,
+      redirect,
+      setAdminAccess,
+      ...otherProps
+    }) => {
+      if (authorized && accountRole === ADMINISTRATOR) {
+        !isAdminAccess && setAdminAccess();
```

Comment:
I think it's not the best place to set flags. Also see comment about `isAdminAccess` above